Packs/RiskIQDigitalFootprint/Integrations/RiskIQDigitalFootprint/RiskIQDigitalFootprint.py

```diff
@@ -0,0 +1,2451 @@
+                                    ' Low, None. This argument supports a single value only.',
+    'INVALID_REMOVED_STATE_ERROR': 'The given value for removed state is invalid. Valid value: Dismissed.'
+                                   ' This argument supports a single value only.',
+    'INVALID_ACTION_ERROR': 'The given value for action is invalid. Valid values: Add, Remove, Update.'
+                            ' This argument supports a single value only.',
+    'INVALID_ARGUMENTS_GET_ASSET': 'Argument uuid cannot be used with other arguments except global and recent.',
+    'MISSING_REQUIRED_KEYS_FOR_GET_ASSET': 'Required argument(s) uuid or [name, type] to get asset details. One or more'
+                                           ' of them are not present.'
+}
+VALID_ASSET_TYPES = ['DOMAIN', 'HOST', 'IP_ADDRESS', 'IP_BLOCK', 'ASN', 'PAGE', 'SSL_CERT', 'CONTACT']
+VALID_ASSET_DETAIL_TYPES = ['SELF_HOSTED_RESOURCE', 'THIRDPARTY_HOSTED_RESOURCE']
+VALID_ASSET_TYPE_MEASURES = ['Added', 'Removed']
+VALID_ASSET_DETAIL_TYPE_MEASURES = ['Added', 'Changed']
+ASSET_TYPE_PLURAL_HR = {
+    'DOMAIN': 'Domains',
+    'HOST': 'Hosts',
+    'IP_ADDRESS': 'IP Addresses',
+    'IP_BLOCK': 'IP Blocks',
+    'AS': 'ASNs',
+    'PAGE': 'Pages',
+    'SSL_CERT': 'SSL Certs',
+    'CONTACT': 'Contacts'
+}
+VALID_RANGES = ['1', '7', '30']
+ASSET_TYPE_HR = {
+    'DOMAIN': 'Domain',
+    'HOST': 'Host',
+    'IP_ADDRESS': 'IP Address',
+    'IP_BLOCK': 'IP Block',
+    'AS': 'ASN',
+    'PAGE': 'Page',
+    'SSL_CERT': 'SSL Cert',
+    'CONTACT': 'Contact',
+    'SELF_HOSTED_RESOURCE': 'Self Hosted Resource',
+    'THIRDPARTY_HOSTED_RESOURCE': 'ThirdParty Hosted Resource'
+}
+DEEP_LINK_PREFIX = {
+    'ASSET_CHANGES_SUMMARY': 'https://app.riskiq.net/a/main/index#/dashboard/inventorychanges',
+    'ASSET_CHANGES': 'https://app.riskiq.net/a/main/index#/dashboard/inventorychanges/details'
+}
+INCOMPLETE_TASK_STATES = ['PENDING', 'RUNNING']
+COMPLETE_TASK_STATES = ['COMPLETE', 'INCOMPLETE', 'FAILED', 'WARNING']
+VALID_ASSET_STATES = ['Candidate', 'Approved Inventory', 'Requires Investigation', 'Dependencies', 'Monitor Only']
+VALID_ASSET_PRIORITY = ['High', 'Medium', 'Low', 'None']
+VALID_ACTIONS = ['ADD', 'REMOVE', 'UPDATE']
+
+
+class Client(BaseClient):
+    """
+    Client to use in integration with powerful http_request.
+    It extends the base client and uses the http_request method for the API request.
+    Handle some exceptions externally.
+    """
+
+    def __init__(self, base_url: str, verify: bool, proxy: bool, auth: Tuple[str, str], request_timeout: int):
+        super().__init__(base_url=base_url, verify=verify, proxy=proxy, auth=auth)
+        self.request_timeout = request_timeout
+
+    def populate_error_message_according_to_status_code(self, resp, status_code):
+        if status_code == 407:
+            demisto.info('Response code: {}. {}'.format(status_code, MESSAGES['PROXY_ERROR']))
+            raise ValueError(MESSAGES['PROXY_ERROR'])
```

Comment:
@avidan-H We have raised this ValueError to give a more user-friendly error message and we verified about this approach in the demisto-developers slack channel as well which can be verified by clicking on this [link](https://dfircommunity.slack.com/archives/CMWN08YB1/p1591368605171700)

Reply:
I understand wanting to provide a cleaner error output to the user rather than a dense exception stack trace and I'm all for it. My objection is to masking the original error (if there is one) with a more generic exception type like `ValueError` (or raising a general exception when a more pointed one would be more useful), which can be misleading for the user (or developer, should the need arise to debug) as to what the true issue is. You could address this by implementing specific exceptions as I suggested in [my comment](https://github.com/demisto/content/pull/8300/files#r466536013).